src/Symfony/Component/Validator/Constraints/UrlValidator.php

```diff
@@ -26,7 +26,7 @@ class UrlValidator extends ConstraintValidator
             (%s)://                                 # protocol
             (((?:[\_\.\pL\pN-]|%%[0-9A-Fa-f]{2})+:)?((?:[\_\.\pL\pN-]|%%[0-9A-Fa-f]{2})+)@)?  # basic auth
             (
-                ([\pL\pN\pS\-\_]+\.)*(([\pL\pN]|xn\-\-[\pL\pN-]+)+\.?) # a domain name
+                ([\pL\pN\pS\-\_]+\.?)*(([\pL\pN]|xn\-\-[\pL\pN-]+)+\.?) # a domain name
```

Comment:
making this dot option makes this regex subject to ReDoS, as it will need to backtrack exponentially in case of non-match, due to the nested repetitions that can now be satisfied by any shorer length for the inner repetition.

to me, the right fix is rather to fix the second part of the line, adding `-` and `_` as allowed in the `[\pL\pN]` we have right now.